src/path.ts

```diff
@@ -1,8 +1,9 @@
+import { RemoveIndex } from './utils';
 import { PathParameters } from './pathParameters';
 import { ClientConfiguration } from './configuration';
 
 export type Path<Configuration extends ClientConfiguration> = <
-  P extends keyof Configuration
+  P extends keyof RemoveIndex<Configuration>
```

Comment:
Removing index type here unlocks autocompletion and results in a type error when a user wants to access a non-existent path